docs/index-management/index-performance.md

```diff
@@ -466,13 +466,51 @@ increases. Often, such traversals can be rewritten as constrained
 traversals that can utilize a vertex-centric index to ensure acceptable
 performance at scale.
 
+### Vertex-centric indexes on adjacent vertex ids
+
+In some cases it is relevant to find an edge based on properties of the adjacent vertex.
+Let's say we want to find out whether or not Hercules has battled Cerberus.
+```groovy
+h = g.V().has('name', 'hercules').next()
+g.V(h).out('battled').has('name', 'cerberus').limit(1).hasNext()
+```
+
+A query like this can not use a vertex centric index because it filters on vertex properties rather than edge properties.
+But by restructuring the query, we can achieve exactly this.
+As both vertices are known, the vertex ids can be used to select the edge.
+In contrast to the name "Cebereus", which is a property of the adjacent vertex, the id of this vertex is already saved within the connecting edge itself.
+Therefore, this query runs much faster if hercules has battled many opponents:
+```groovy
+h = g.V().has('name', 'hercules').next()
+c = g.V().has('name', 'cerberus').next()
+g.V(h).outE('battled').where(inV().is(c)).limit(1).hasNext()
+```
+
+Assuming there is a global index on the 'name' property, this already improves the performance a lot, because it's not necessary to fetch every adjacent vertex anymore.
+But analog to other edge properties, a vertex-centric index can be used to further improve the performance:
+```groovy
+graph.tx().rollback()  //Never create new indexes while a transaction is active
```

Comment:
Not necessary to build this index explicitly. It is available automatically from storage layout. See column part of https://docs.janusgraph.org/advanced-topics/data-model/#individual-edge-layout

Reply:
By 'this index' you mean the index on the adjacent ids?
I see that the storage backend can efficiently retrieve edges by their column. But the column is composed of multiple values with only one of them being the adjacent id. I don't know how it would be possible to access edges by only a part of their column.

Reply:
Alright, I had a closer look at it and I admit I somehow skipped performance testing index-aware queries without explicitly creating these indexes.
As a consequence, I don't see any point in [this](https://github.com/JanusGraph/janusgraph/pull/1937#issuecomment-585091811) functionality anymore.